.github/workflows/rust.yml

```diff
@@ -23,139 +23,112 @@ jobs:
-        #rv: [1.58.1, stable, beta, nightly]
-        rv: [nightly]
     steps:
-      - name: Install Dependencies
-        if: startsWith(matrix.os, 'Ubuntu')
-        run: |
-          sudo apt-get update
-          sudo apt-get install -y build-essential clang ocl-icd-opencl-dev
       - name: Checkout Sources
         uses: actions/checkout@v3
-      - name: Rust Cache
-        uses: Swatinem/rust-cache@v2
-        timeout-minutes: ${{ fromJSON(env.CACHE_TIMEOUT_MINUTES) }}
-        continue-on-error: true
-      - name: Cargo Build
-        run: cargo build --profile dev
+      - name: Setup sccache
+        uses: hanabi1224/sccache-action@v1.0.0
+        with:
+          # hard code release-name for macos, it always get rate limited when calling github api
+          release-name: v0.3.0
+          cache-key: ${{ runner.os }}-sccache-${{ hashFiles('rust-toolchain.toml') }}
+          cache-save: ${{ github.event_name != 'pull_request' }}
+      - name: Install Apt Dependencies
+        if: startsWith(matrix.os, 'Ubuntu')
+        run: |
+          lscpu # the job may run on different CPUs, list cpu here for analysing build time
+          sudo make install-deps
+      - name: Cargo Check
+        run: cargo check --timings
+        env:
+          CC: "sccache clang"
+          CXX: "sccache clang++"
+      - uses: actions/upload-artifact@v3
+        with:
+          name: build-timings
+          path: |
+            target/cargo-timings/*
+          if-no-files-found: error
 
   calibnet-check:
     name: Calibnet sync check
     runs-on: buildjet-8vcpu-ubuntu-2004
     timeout-minutes: 30
     steps:
-      - name: Install Dependencies
-        run: |
-          sudo apt-get update
-          sudo apt-get install -y build-essential clang ocl-icd-opencl-dev
       - name: Checkout Sources
         uses: actions/checkout@v3
         with:
           submodules: true
-      - name: Rust Cache
-        uses: Swatinem/rust-cache@v2
-        timeout-minutes: ${{ fromJSON(env.CACHE_TIMEOUT_MINUTES) }}
-        continue-on-error: true
+      - name: Apt Dependencies
+        run: sudo make install-deps
+      - name: Setup sccache
+        uses: hanabi1224/sccache-action@v1.0.0
+        with:
+          cache-key: ${{ runner.os }}-sccache-test-${{ hashFiles('rust-toolchain.toml') }}
+          cache-save: ${{ github.event_name != 'pull_request' }}
       - name: build and install binaries
         run: make install
+        env:
+          CC: "sccache clang"
+          CXX: "sccache clang++"
       - name: download snapshot
         run: forest-cli --chain calibnet chain fetch -s /tmp/snapshots/
       - name: import snapshot and run Forest
         run: |
           forest --chain calibnet --target-peer-count 50 --encrypt-keystore false --halt-after-import --import-snapshot /tmp/snapshots/*.car
           forest --chain calibnet --target-peer-count 50 --encrypt-keystore false --detach
-      - name: wait for sync and check health
```

Comment:
This stops working, disable for now?

Reply:
@hanabi1224 you just need to :top: :fist_right: :heavy_check_mark: this guys https://github.com/ChainSafe/forest/pull/2093

Reply:
...and rebase. The reason is, calibnet got a version upgrade yesterday, and we are currently lagging (for various reasons), so we are unable to sync to calibnet at the moment.